src/utl/product_generator.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import yaml
+import os
+import json
+
+from utl.webscraper import WebScraper
+from utl.openaillm import OpenAIClient
+from bgsexception import BgsException
+
+logger = logging.getLogger(__name__)
+
+def _create_description(vendor, model, url: str):
+
+    scraper = WebScraper(url, depth=0)
+    (actual_results, links) = scraper.scrape_single_page(url)
+    actual_results = actual_results[:1000]
+
+    # Ignore vendor for now, and just use OpenAI
+    temperature = 0
+    timeout = 100
+    max_retries = 3
+    llm = OpenAIClient(model=model)
+
+    query = (
+        "Provide a summary for a high level business product "
+        "which details what data can be found "
+    )
+    description = llm.send_prompt(query, actual_results)
+
+    return description
+
+def create_product_file(
+        output_dir: str, file_name: str, namespace: str, name: str,
+        tags: str, description: str, url: str, vendor: str, model: str):
+
+    if not description:
+        description = _create_description(vendor, model, url)
```

Comment:
ideally shouldn't there be a more explicit switch to use ai, instead of AI being the default option. A more explicit argument like --ai-description would work better, and then just throw an error if both description and --ai-description are missing.